writer.go

```diff
@@ -11,27 +12,42 @@ type writer struct {
 	color ColorOption
 }
 
-func newWriter(w io.Writer, color ColorOption) *writer {
-	return &writer{w: w, color: color}
+func newWriter(out io.Writer, color ColorOption) *writer {
+	w := &writer{w: out, color: color}
+	withColor(w)
+	return w
 }
 
 func (w *writer) Flush(level Level) (err error) {
-	var unwritten = w.b.Bytes()
+	msg := w.b.Bytes()
 
 	if w.color != ColorOff {
-		color := _levelToColor[level]
-		unwritten = []byte(color.Sprintf("%s", unwritten))
+		msg = _levelToColor[level](msg)
 	}
 
 	if lw, ok := w.w.(LevelWriter); ok {
-		_, err = lw.LevelWrite(level, unwritten)
+		_, err = lw.LevelWrite(level, msg)
 	} else {
-		_, err = w.w.Write(unwritten)
+		_, err = w.w.Write(msg)
 	}
 	w.b.Reset()
 	return err
 }
 
+var _levelToColor = map[Level]func([]byte) []byte{
+	Debug: newColor(252), // light grey
+	Trace: newColor(82),  // bright green
+	Info:  newColor(27),  // blue
+	Warn:  newColor(226), // yellow
+	Error: newColor(196), // red
```

Comment:
Where did these numbers come from?

Reply:
Thse are ansi 8-bit color codes , from https://en.wikipedia.org/wiki/ANSI_escape_code#8-bit

I guess it would be good to include that information in a comment.

Reply:
`termcap`/`terminfo` is unreliable (because it depends on the user having correctly defined `TERM`, and the terminfo being available on the machine where the code runs which might not be the machine where the user has his/her terminal app.

Anyway, this is 2022. Does anyone still use a terminal app that supports only 8 colors?